### PR TITLE
WIP Added initial framework for checking spelling in docs

### PR DESCRIPTION
**Pull request by astrofrog**
This is a starting point for addressing https://github.com/astropy/astropy/issues/7752 - it makes it possible to run:

```
make spelling
```

and it adds a whitelist of list of words, which currently includes all words considered misspelt by the spell checker. The idea is that someone needs to go through the ``spelling_wordlist.txt`` and remove incorrectly spelt words and fix them in the docs at the same time (note that this also spell checks docstrings). Once only valid words remain in ``spelling_wordlist.txt``, then we could eventually merge this PR. We also need to add a ``make spelling`` step to the docs build in the CI.

Note that this functionality requires the sphinxcontrib-spelling package to be installed.

Just for info, there is a known issue with words like "didn't" - https://github.com/sphinx-contrib/spelling/issues/22

I don't have time to actually work on fixing the spelling, but people should feel free to add commits to this PR, or start a new PR and cherry-pick the commit here.

### CODE REVIEW

**Comment by mhvk** [comment]
This drifted up and really is quite nice, though it would help to exclude the list of contributors a priori -- most of the words seem to be names of people.

File: docs/spelling_wordlist.txt [hunk 1]
```diff
@@ -0,0 +1,1724 @@
...
+ddof
+de
+dec
+deca
+deci
+decimalyear
+deepcopied
+deepcopy
+defaultvalue
+defind
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 2]
```diff
@@ -0,0 +1,1724 @@
...
+definitons
+deka
+deleteable
+deluxetable
+delval
+denywrite
+dependending
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 3]
```diff
@@ -0,0 +1,1724 @@
...
+deriv
+deserialization
+desribed
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 4]
```diff
@@ -0,0 +1,1724 @@
...
+destdir
+det
+deteremined
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 5]
```diff
@@ -0,0 +1,1724 @@
...
+dev
+dex
+df
+di
+diaconis
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 6]
```diff
@@ -0,0 +1,1724 @@
...
+dict
+dicts
+didn
+diff
+diffing
+diffs
+dimShape
+dir
+discontiguous
+discrepent
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 7]
```diff
@@ -0,0 +1,1724 @@
...
+discretize
+distlimit
+distmod
+dms
+docstring
+docstrings
+doctest
+doctestplus
+doctests
+doesn
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 8]
```diff
@@ -0,0 +1,1724 @@
...
+doi
+doppler
+doubleComplex
+downsample
+dropax
+dropdown
+ds
+dsiplayed
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 9]
```diff
@@ -0,0 +1,1724 @@
...
+dst
+dt
+dtype
+dtypes
+dy
+dytpe
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 10]
```diff
@@ -0,0 +1,1724 @@
...
+eb
+ecl
+ecsv
+edu
+eff
+efunc
+eg
+einsum
+electronvolts
+emittence
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 11]
```diff
@@ -0,0 +1,1724 @@
...
+equivalance
+equivalencies
+equivalency
+erfa
+errorbar
+exa
+exbi
+exc
+exoplanets
+exsiting
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 12]
```diff
@@ -0,0 +1,1724 @@
...
+extname
+extnum
+extremly
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 13]
```diff
@@ -0,0 +1,1724 @@
...
+followup
+formatwarning
+formulae
+formulæ
+fortran
+fpp
+fread
+freetype
+frequentist
+frequestly
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 14]
```diff
@@ -0,0 +1,1724 @@
...
+fromstring
+fromsys
+fromunit
+fset
+ftp
+func
+funcs
+functons
+funpack
+futher
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 15]
```diff
@@ -0,0 +1,1724 @@
...
+fromcoord
+fromfile
+fromstring
+fromsys
+fromunit
+fset
+ftp
+func
+funcs
+functons
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 16]
```diff
@@ -0,0 +1,1724 @@
...
+hashable
+hashorurl
+hbar
+hdf
+hdrLoc
+hdu
+hdulist
+heapsort
+hecto
+heirarchical
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 17]
```diff
@@ -0,0 +1,1724 @@
...
+html
+htmldict
+http
+hypergeometric
+iau
+icrs
+idx
+ie
+iers
+ignorning
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 18]
```diff
@@ -0,0 +1,1724 @@
...
+iloc
+im
+inA
+inB
+indendation
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 19]
```diff
@@ -0,0 +1,1724 @@
...
+initializable
+initval
+initvals
+inline
+inorder
+inplace
+inputter
+instancemethod
+instanciation
+inteded
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 20]
```diff
@@ -0,0 +1,1724 @@
...
+intercompatibility
+intereted
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 21]
```diff
@@ -0,0 +1,1724 @@
...
+matplotlib
+matplotlibrc
+maxerr
+maxima
+maxiter
+maxw
+mebi
+memmap
+memmaped
+memoized
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 22]
```diff
@@ -0,0 +1,1724 @@
...
+milli
+milliarcsecond
+milliarcseconds
+minerr
+mininium
+minmax
+minpad
+mins
+minversion
+misssing
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 23]
```diff
@@ -0,0 +1,1724 @@
...
+mpl
+msg
+mtype
+muB
+mul
+multicol
+multiindex
+multithreaded
+multithreading
+mutliple
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 24]
```diff
@@ -0,0 +1,1724 @@
...
+mvoid
+namedtuples
+namespace
+namse
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 25]
```diff
@@ -0,0 +1,1724 @@
...
+ndarrays
+nddata
+ndim
+nelem
+newfile
+newrepr
+nmi
+nnn
+noninformative
+normaliation
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 26]
```diff
@@ -0,0 +1,1724 @@
...
+num
+numarray
+numdiffs
+numpy
+numr
+nyquist
+objfunc
+obstime
+obstimes
+occuring
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 27]
```diff
@@ -0,0 +1,1724 @@
...
+octalRegex
+odict
+offline
+ohser
+oir
+ommiting
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 28]
```diff
@@ -0,0 +1,1724 @@
...
+ommitted
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 29]
```diff
@@ -0,0 +1,1724 @@
...
+ording
+org
+orthomorphic
+ostream
+ouput
+outlier
+outputter
+overplot
+overridable
+overriden
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 30]
```diff
@@ -0,0 +1,1724 @@
...
+poisson
+polyconic
+pos
+posvel
+powerlaws
+pprint
+pprinting
+prasing
+pre
+predections
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 31]
```diff
@@ -0,0 +1,1724 @@
...
+unicodeChar
+uninstall
+uninstalled
+unintutive
+uniq
+unitless
+unix
+unkown
+unmapping
+unneccesary
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 32]
```diff
@@ -0,0 +1,1724 @@
...
+unpickling
+unrounded
+unsignedByte
+unsignedInt
+unstage
+unsubscribe
+unsubscribed
+unsymmetric
+upcast
+updation
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 33]
```diff
@@ -0,0 +1,1724 @@
...
+urls
+useable
+useblanks
+usecPickle
+usecase
+username
+ut
+utc
+util
+utilies
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 34]
```diff
@@ -0,0 +1,1724 @@
...
+utils
+utype
+validator
+vals
+ve
+velref
+vendored
+ver
+verbotten
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]: :)
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 35]
```diff
@@ -0,0 +1,1724 @@
...
+whitelisted
+wi
+wien
+wildcard
+wildcarded
+wildcards
+wilson
+wis
+workflow
+wouldn
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 36]
```diff
@@ -0,0 +1,1724 @@
...
+voxel
+vspan
+vstack
+vstep
+vtest
+wCDM
+wa
+waCDM
+wald
+wasn
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 37]
```diff
@@ -0,0 +1,1724 @@
...
+wcsaxes
+wcsbth
+wcskey
+wcslib
+wcspih
+wcsprm
+webpage
+weigth
+weren
+whiich
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 38]
```diff
@@ -0,0 +1,1724 @@
...
+wcs
+wcsaxes
+wcsbth
+wcskey
+wcslib
+wcspih
+wcsprm
+webpage
+weigth
+weren
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestio
```

File: docs/spelling_wordlist.txt [hunk 39]
```diff
@@ -0,0 +1,1724 @@
...
+wb
+wcs
+wcsaxes
+wcsbth
+wcskey
+wcslib
+wcspih
+wcsprm
+webpage
+weigth
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 40]
```diff
@@ -0,0 +1,1724 @@
...
+Zw
+Zé
+aBc
+aa
+aastex
+abc
+abcoulomb
+abosolute
+accessor
+acessing
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

File: docs/spelling_wordlist.txt [hunk 41]
```diff
@@ -0,0 +1,1724 @@
...
+Zscale
+Zucker
+Zw
+Zé
+aBc
+aa
+aastex
+abc
+abcoulomb
+abosolute
```
Comment (bsipocz) [inline docs/spelling_wordlist.txt]:
```suggestion
```

**Comment by bsipocz** [comment]
@astrofrog - reviving this, or any other approach to have a running spell checker would be very valuable I think.

**Pull request closed without merging**